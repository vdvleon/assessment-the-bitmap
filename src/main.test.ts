import { WritableStreamBuffer } from "stream-buffers";
import { join } from "path";
import { createReadStream } from "fs";
import main from "./main";

describe("main", () => {
    it("should write the correct output for the input", async () => {
        const inputStream = createReadStream(
            join(__dirname, "..", "resources", "exampleInput.txt")
        );
        const outputStream = new WritableStreamBuffer();
        const errorStream = new WritableStreamBuffer();

        await main(inputStream, outputStream, errorStream);

        expect(outputStream.getContentsAsString()).toMatchSnapshot();
        expect(errorStream.getContentsAsString()).toMatchSnapshot();
    });
});
