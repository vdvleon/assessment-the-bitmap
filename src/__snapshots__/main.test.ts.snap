// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should write the correct output for the input 1`] = `
"0 1 2 3 4 4 3 2 1 0
1 0 1 2 3 3 2 1 0 1
2 1 0 1 2 2 1 0 1 2
3 2 1 0 1 1 0 1 2 3
4 3 2 1 0 0 1 2 3 4
4 3 2 1 0 0 1 2 3 4
3 2 1 0 1 1 0 1 2 3
2 1 0 1 2 2 1 0 1 2
1 0 1 2 3 3 2 1 0 1
0 1 2 3 4 4 3 2 1 0
0 0 0 0
0 1 1 0
0 1 1 0
0 0 0 0
âˆž
0
8 7 6 5 4 4 5 6 7 8
7 6 5 4 3 3 4 5 6 7
6 5 4 3 2 2 3 4 5 6
5 4 3 2 1 1 2 3 4 5
4 3 2 1 0 0 1 2 3 4
4 3 2 1 0 0 1 2 3 4
5 4 3 2 1 1 2 3 4 5
6 5 4 3 2 2 3 4 5 6
7 6 5 4 3 3 4 5 6 7
8 7 6 5 4 4 5 6 7 8
"
`;

exports[`main should write the correct output for the input 2`] = `false`;
